ios/Swift/imagepicker/ImagePickerViewController.swift

```diff
@@ -64,78 +64,84 @@ extension ViewController {
-            } else {
-                // Parse the response
-                print(json)
-                let responses: JSON = json["responses"][0]
+            do {
+                let json = try JSON(data: dataToParse)
+                let errorObj: JSON = json["error"]
                 
-                // Get face annotations
-                let faceAnnotations: JSON = responses["faceAnnotations"]
-                if faceAnnotations != nil {
-                    let emotions: Array<String> = ["joy", "sorrow", "surprise", "anger"]
-                    
-                    let numPeopleDetected:Int = faceAnnotations.count
-                    
-                    self.faceResults.text = "People detected: \(numPeopleDetected)\n\nEmotions detected:\n"
-                    
-                    var emotionTotals: [String: Double] = ["sorrow": 0, "joy": 0, "surprise": 0, "anger": 0]
-                    var emotionLikelihoods: [String: Double] = ["VERY_LIKELY": 0.9, "LIKELY": 0.75, "POSSIBLE": 0.5, "UNLIKELY":0.25, "VERY_UNLIKELY": 0.0]
+                self.spinner.stopAnimating()
+                self.imageView.isHidden = true
+                self.labelResults.isHidden = false
+                self.faceResults.isHidden = false
+                self.faceResults.text = ""
+                
+                // Check for errors
+                if (errorObj.dictionaryValue != [:]) {
+                    self.labelResults.text = "Error code \(errorObj["code"]): \(errorObj["message"])"
+                } else {
+                    // Parse the response
+                    print(json)
+                    let responses: JSON = json["responses"][0]
                     
-                    for index in 0..<numPeopleDetected {
-                        let personData:JSON = faceAnnotations[index]
+                    // Get face annotations
+                    let faceAnnotations: JSON = responses["faceAnnotations"]
+                    if faceAnnotations != JSON.null {
+                        let emotions: Array<String> = ["joy", "sorrow", "surprise", "anger"]
+                        
+                        let numPeopleDetected:Int = faceAnnotations.count
+                        
+                        self.faceResults.text = "People detected: \(numPeopleDetected)\n\nEmotions detected:\n"
                         
-                        // Sum all the detected emotions
-                        for emotion in emotions {
-                            let lookup = emotion + "Likelihood"
-                            let result:String = personData[lookup].stringValue
-                            emotionTotals[emotion]! += emotionLikelihoods[result]!
+                        var emotionTotals: [String: Double] = ["sorrow": 0, "joy": 0, "surprise": 0, "anger": 0]
+                        var emotionLikelihoods: [String: Double] = ["VERY_LIKELY": 0.9, "LIKELY": 0.75, "POSSIBLE": 0.5, "UNLIKELY":0.25, "VERY_UNLIKELY": 0.0]
```

Comment:
This is a long line continuation, maybe break up as:

                            var emotionLikelihoods: [String: Double] = [
                                "VERY_LIKELY": 0.9, "LIKELY": 0.75,"POSSIBLE": 0.5, "UNLIKELY":0.25, "VERY_UNLIKELY": 0.0]

Which just fits under 120 cols